Lib/test/test_itertools.py

```diff
@@ -1183,7 +1183,7 @@ def test_pairwise(self):
             pairwise(None)                                  # non-iterable argument
 
     # TODO: RUSTPYTHON
-    @unittest.skip("TODO: RUSTPYTHON, hangs")
+    @unittest.expectedFailure
     def test_pairwise_reenter(self):
```

Comment:
I've tried to make this test to pass as well, but for some reason when it's reentering the iter `old` gets reset to None, causing the next `old` to advance the iter one more time than needed.
So for the first assert:
```
Expected: ([1], ([1], [3]))
Got: ([1], ([3], [4]))
```

If someone can guide me where to investigate it would be great:)